### ISSUE TITLE
Direct Code Editor Bugs

### ISSUE THREAD

**Issue by TanTai97o** [Sketchware-Pro/Sketchware-Pro#1467]
error applying background color when adding transparency to example #002196F3

Video:
https://streamable.com/1lyd09

v6.4.0-SNAPSHOT-0d3ccb6-minApi26

**Comment by elfilibusterismo** [comment]
It's not exactly a bug, as this thing already existed before. Sketchware strictly uses the #%06X format, which supports only #RRGGBB colors, excluding alpha transparency (#AARRGGBB).

**Comment by TanTai97o** [comment]
> It's not exactly a bug, as this thing already existed before. Sketchware strictly uses the #%06X format, which supports only #RRGGBB colors, excluding alpha transparency (#AARRGGBB).

But when I add #002196F3 in direct code editor and save, it only shows #2196F3 and when I save again, the background color does not change to blue.

**Comment by elfilibusterismo** [comment]
> > It's not exactly a bug, as this thing already existed before. Sketchware strictly uses the #%06X format, which supports only #RRGGBB colors, excluding alpha transparency (#AARRGGBB).
> 
> But when I add #002196F3 in direct code editor and save, it only shows #2196F3 and when I save again, the background color does not change to blue.

Do u even understand my comment.

Anyways @hasrat-ali what do you think? Should we change this in a.a.a.Ox?

**Comment by TanTai97o** [comment]
> > > It's not exactly a bug, as this thing already existed before. Sketchware strictly uses the #%06X format, which supports only #RRGGBB colors, excluding alpha transparency (#AARRGGBB).
> > 
> > But when I add #002196F3 in direct code editor and save, it only shows #2196F3 and when I save again, the background color does not change to blue.
> 
> Do u even understand my comment.
> 
> Anyways @hasrat-ali what do you think? Should we change this in a.a.a.Ox?

I'm not good English so I don't really understand your comment

**Comment by hasrat-ali** [comment]
> Anyways @hasrat-ali what do you think? Should we change this in a.a.a.Ox?

We need to do that, but we should also handle color redundancy too. e.g. `#008dcd` for `#008dcd` not `#ff008dcd`

**Comment by hasrat-ali** [comment]
```java
public static String formatColor(int color) {
    int alpha = (color >> 24) & 0xFF;
    
    if (alpha != 0xFF) {
        return String.format("#%08X", color);
    } else {
        return String.format("#%06X", color);
    }
}
```

**Comment by elfilibusterismo** [comment]
So, uh, are you going to do it?

**Comment by hasrat-ali** [comment]
Well, i didn't get enough time. Can you do it instead?

**Comment by hasrat-ali** [comment]
Fixed with 744df05688e6dda398819f39d8c55f7efff73784
